test/core.js

```diff
@@ -1134,6 +1134,57 @@ describe('Core', () => {
             expect(res.result).to.be.true();
         });
 
+        it('sets `request.isInjected = true` for requests created via `server.inject`', async () => {
+
+            const server = Hapi.server();
+            server.route({ method: 'GET', path: '/', handler: (request) => request.isInjected });
+
+            const options = {
+                url: '/'
+            };
+
+            const res = await server.inject(options);
+            expect(res.statusCode).to.equal(200);
+            expect(res.result).to.be.true();
+        });
+
+        it('`request.isInjected` access is read-only', async () => {
+
+            const server = Hapi.server();
+            let request = null;
+            server.route({ method: 'GET', path: '/', handler: (_request) => {
+
+                request = _request; // hoist request object to perform illegal assignment assertion
```

Comment:
This is a little convoluted. You can just move the assertion inline here:

```js
expect(() => {
  request.isInjected = false;
}).to.throw(...);
```

Reply:
Sure, I reverted to the previous commit where I did inline assertion, I'm not sure if you saw my previous comment but my reasoning to do the above was to make failed assertion clearer.  Consider this assertion:

```
expect(illegalAssignment).to.throw('The wrong thing');
```

would produce:

<img width="474" alt="Screen Shot 2020-07-25 at 8 20 19 PM" src="https://user-images.githubusercontent.com/191729/88469199-64468600-ceb4-11ea-8a6e-cab4e94ee283.png">

So it would fail the test but might not as clear as performing assertion outside route handler.  Happy to do it this way however as the code itself seems to be cleaner.